qmi/instruments/thorlabs/mpc320.py

```diff
@@ -0,0 +1,507 @@
+"""Module for a Thorlabs MPC320 motorised fibre polarisation controller."""
+
+from dataclasses import dataclass
+import logging
+from typing import Dict, List
+from qmi.core.context import QMI_Context
+from qmi.core.exceptions import QMI_InstrumentException
+from qmi.core.instrument import QMI_Instrument, QMI_InstrumentIdentification
+from qmi.core.rpc import rpc_method
+from qmi.core.transport import create_transport
+from qmi.instruments.thorlabs.apt_packets import (
+    HW_GET_INFO,
+    MOD_GET_CHANENABLESTATE,
+    MOT_GET_USTATUSUPDATE,
+    MOT_MOVE_ABSOLUTE,
+    MOT_MOVE_COMPLETED,
+    MOT_MOVE_HOMED,
+    MOT_SET_EEPROMPARAMS,
+    POL_GET_SET_PARAMS,
+)
+from qmi.instruments.thorlabs.apt_protocol import (
+    AptChannelJogDirection,
+    AptChannelState,
+    AptMessageId,
+    AptProtocol,
+)
+
+# Global variable holding the logger for this module.
+_logger = logging.getLogger(__name__)
+
+
+@dataclass
+class Thorlabs_MPC320_Status:
+    """
+    Data class for the status of the MPC320
+
+    Attributes:
+        channel:        Channel number.
+        position:       Absolute position of the channel in degrees.
+        velocity:       Velocity in controller units.
+        motor_current:  Current of motor in mA
+    """
+
+    channel: int
+    position: float
+    velocity: int
+    motor_current: int
+
+
+@dataclass
+class Thorlabs_MPC320_PolarisationParameters:
+    """
+    Data class for the polarisation parameters of the MPC320
+
+    Attributes:
+        velocity:       The velocity in percentage of the max velocity 400 degrees/s.
+        home_position:  The home position of all the paddles/channels in degrees.
+        jog_step1:      The position to move paddel/channel 1 by for a jog step in degrees.
+        jog_step2:      The position to move paddel/channel 2 by for a jog step in degrees.
+        jog_step3:      The position to move paddel/channel 3 by for a jog step in degrees.
+    """
+
+    velocity: float
+    home_position: float
+    jog_step1: float
+    jog_step2: float
+    jog_step3: float
+
+
+Thorlabs_MPC320_ChannelMap: Dict[int, int] = {1: 0x01, 2: 0x02, 3: 0x04}
+
+
+class Thorlabs_MPC320(QMI_Instrument):
+    """
+    Driver for a Thorlabs MPC320 motorised fibre polarisation controller.
+    """
+
+    DEFAULT_RESPONSE_TIMEOUT = 1.0
+
+    # the maximum range for a paddle is 170 degrees
+    # the value returned by the encoder is 1370 for 170 degrees
+    ENCODER_CONVERSION_UNIT = 170 / 1370
+
+    MIN_POSITION_DEGREES = 0
+    MAX_POSITION_DEGREES = 170
+
+    MIN_VELOCITY_PERC = 10
+    MAX_VELOCITY_PERC = 100
```

Comment:
You are using `int`s here, could they also be floats?

Reply:
they can only be ints